docs/architecture/fsm-1.md

```diff
@@ -0,0 +1,78 @@
+# Connection Handshake FSM
+
+
+## Input Events:
+
```

Comment:
We should fill this in. The events into the FSM that are relevant are:
- the chain event that creates the FSM (`openInit`, `openTry` or `openAck`)
- response to LC query
- response to connection query
- other chain events:
  - new block
  - on-chain client updated

All the others will be dealt with/ filtered by the event handler